evm/p2p/peer.py

```diff
@@ -39,6 +44,27 @@
 )
 
 
+@asyncio.coroutine
+def handshake(remote, privkey):
+    """Perform the auth and P2P handshakes with the given remote.
+
+    Return a Peer connected to that remote in case both handshakes are successful and at least one
+    of the sub-protocols supported by us is also supported by the remote.
+
+    Raises UnreachablePeer if we cannot connect to the peer or UselessPeer if none of the
+    sub-protocols supported by us is also supported by the remote.
+    """
+    try:
+        peer = yield from auth.handshake(remote, privkey)
+        msg = yield from peer.read_msg()
+    except (ConnectionRefusedError, OSError) as e:
```

Comment:
Are either of these exceptions specific to the lines in the `try` block or can both lines raise either exception?  If it's the former, can we split this into separate `try` statements?  

I suspect it's the later but since I can't find anything else in this PR to leave feedback on :)

Reply:
Hmm, good catch; I'm pretty sure only the open_connection() call in auth.handshake() can raise those, so I'll move the read_msg() out of the try block